ingest-app/src/cmr/ingest/api/subscriptions.clj

```diff
@@ -224,12 +248,13 @@
                             content-type
                             headers)
           native-id (get-unique-native-id request-context tmp-subscription)
-          new-subscription (assoc  tmp-subscription :native-id native-id)
-          newer-subscription (add-fields-if-missing request-context new-subscription)
-          subscriber-id (get-subscriber-id newer-subscription)]
+          sub-with-native-id (assoc  tmp-subscription :native-id native-id)
+          final-sub (add-fields-if-missing request-context sub-with-native-id)
+          subscriber-id (get-subscriber-id final-sub)]
+      (validate-query request-context final-sub)
```

Comment:
Move the `validate-query` call below `check-ingest-permission` call.

Reply:
I like this change - my original thought process was that if the query is bad, it's not worth wasting time checking anything else, so we look for that first. But the same is true if the user has no permissions, which is cheaper to check, so I agree it should go first.